### PR TITLE
 parserdb: Implement RelationFieldWalker → RelationWalker 

### PR DESCRIPTION
**Pull request by tomhoule**
This is going to be required for better autocompletion on relation arguments.

### CODE REVIEW

File: libs/datamodel/parser-database/src/relations.rs [hunk 1]
```diff
@@ -17,6 +17,14 @@ pub(super) fn infer_relations(ctx: &mut Context<'_>) {
     ctx.db.relations = relations;
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, PartialOrd, Ord, Eq)]
+pub struct RelationId(u32);
```
Comment (pimeys) [inline libs/datamodel/parser-database/src/relations.rs:21]: <3

File: libs/datamodel/parser-database/src/relations.rs [hunk 2]
```diff
@@ -44,7 +44,7 @@ impl RelationId {
 #[derive(Debug, Default)]
 pub(crate) struct Relations<'ast> {
     /// Storage. Private. Do not use directly.
-    pub(super) relations_storage: Vec<Relation<'ast>>,
```
Comment (pimeys) [inline libs/datamodel/parser-database/src/relations.rs:39]: yiss

**Review by pimeys — APPROVED** [review]
It's good and y'all, much better. How are you using the new public interface? Or is it something for the future?

**Comment by tomhoule** [comment]
It's not used yet. I was thinking https://github.com/prisma/prisma-engines/commit/2c9fe99c5c399e1116ea9fa5a266fc93c24a0712 as a first use case (filter out from autocompletions)

**Pull request merged**